File: vitalcore-expo/e2e/pages/DashboardPage.ts
import BasePage from './BasePage';

export default class DashboardPage extends BasePage {
  public get welcomeBanner() { return this.getByTestId('dash-welcome-banner'); }
  public get caloriesCard() { return this.getByTestId('dash-calories-card'); }
  public get waterCard() { return this.getByTestId('dash-water-card'); }
  public get sleepCard() { return this.getByTestId('dash-sleep-card'); }
  public get stepsCard() { return this.getByTestId('dash-steps-card'); }
  public get quickLogWaterBtn() { return this.getByTestId('dash-quicklog-water-btn'); }
  public get quickLogStepsBtn() { return this.getByTestId('dash-quicklog-steps-btn'); }

  public async isDashboardLoaded(): Promise<boolean> {
    return this.isElementDisplayed('dash-calories-card');
  }

  public async quickLogWater(): Promise<void> {
    await this.waitAndClick('dash-quicklog-water-btn');
  }
}

File: test_pages/dashboard_page.py
"""Page Object Model for Dashboard Page (/dashboard)."""

from selenium.webdriver.common.by import By
from test_pages.base_page import BasePage

class DashboardPage(BasePage):
    GREETING = (By.XPATH, "//*[contains(text(), 'Good morning') or contains(text(), 'Good afternoon') or contains(text(), 'Good evening')]")
    CALORIES_CARD = (By.XPATH, "//*[contains(text(), 'Calories') or contains(text(), 'kcal')]")
    HYDRATION_CARD = (By.XPATH, "//*[contains(text(), 'Hydration') or contains(text(), 'Water') or contains(text(), 'ml')]")
    SLEEP_CARD = (By.XPATH, "//*[contains(text(), 'Sleep') or contains(text(), 'hrs')]")
    STEPS_CARD = (By.XPATH, "//*[contains(text(), 'Steps') or contains(text(), 'Activity')]")
    ADD_250ML_BTN = (By.XPATH, "//button[contains(text(), '+250') or contains(text(), '250ml')]")
    ADD_500ML_BTN = (By.XPATH, "//button[contains(text(), '+500') or contains(text(), '500ml')]")
    SIMULATOR_BTN = (By.XPATH, "//button[contains(text(), 'Simulator') or contains(text(), 'Try Simulator')]")

    def open(self):
        self.navigate_to("/dashboard")

    def is_dashboard_loaded(self):
        return self.is_visible(*self.GREETING, timeout=8) or self.is_visible(*self.CALORIES_CARD, timeout=8)
